doc/source/tutorials/add_model.rst

```diff
@@ -0,0 +1,327 @@
+
+Step 1: Add Model Architecture Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Model architectures are defined as configuration presets that specify the structural parameters of your model.
+
+1. **Navigate to the model family's config file:**
+
+   .. code-block:: bash
+
+       src/fairseq2/models/qwen/config.py
+
+2. **Add a new architecture function:**
+
+   .. code-block:: python
+
+       @arch("qwen25_3b_instruct")
+       def qwen25_3b_instruct() -> QwenConfig:
+           """Configuration for Qwen2.5-3B-Instruct model.
+           """
+           config = QwenConfig()
+
+           # Set model dimensions and structure
+           config.model_dim = 2048
+           ...
+
+           return config
+
+   **Key points:**
+
+   - The ``@arch`` decorator registers the configuration with the given name
+   - The function name should match or describe the architecture
+   - Use the appropriate config class for the model family (``QwenConfig`` for Qwen models)
+   - Set all necessary parameters for your specific model variant
+
+Step 2: Create Asset Card
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Asset cards are YAML files that tell fairseq2 where to find your model checkpoints and how to load them.
+
+1. **Navigate to the model family's asset card file:**
+
+    .. code-block:: bash
+
+        src/fairseq2/assets/cards/models/qwen.yaml
+
+2. **Add a new asset card entry:**
+
+    .. code-block:: yaml
+
+        name: qwen25_3b_instruct
+        model_family: qwen
+        model_arch: qwen25_3b
+        checkpoint: "hg://qwen/qwen2.5-3b-instruct"
+        tokenizer: "hg://qwen/qwen2.5-3b-instruct"
+        tokenizer_family: qwen
+        tokenizer_config:
+            use_im_end: true
+
+- ``name``: The model name users will use (e.g., ``load_model("qwen25_3b_instruct")``)
+- ``model_family``: Which model family handles this model (``qwen``)
+- ``model_arch``: Which architecture configuration to use (``qwen25_3b``)
+- ``checkpoint``: Where to download the model weights from
+- ``tokenizer``: Where to download the tokenizer from
+- ``tokenizer_family``: Which tokenizer family to use
+- ``tokenizer_config``: Tokenizer-specific settings
+
+For more details on asset card options, see :ref:`basics-assets`.
+
+Step 3: Verify the Integration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+After adding the configuration and asset card, verify that your model is properly registered:
+
+1. **Check if model is recognized:**
+
+   .. code-block:: bash
+
+       # List all models to see if yours appears
+       python -m fairseq2.assets list --kind model
+
+       # Look specifically for your model
+       python -m fairseq2.assets list --kind model | grep qwen25_3b_instruct
+
+2. **Test model loading:**
+
+   .. code-block:: python
+
+       import fairseq2
+       from fairseq2.models.hub import load_model
+
+       # Test loading your model
+       try:
+           model = load_model("qwen25_3b_instruct")
```

Comment:
Not sure if we should mention it here, but we also have "model hubs" for each model family that exposes methods (like load from path, create a newly initialized, list architectures so on) beyond loading. Maybe makes sense to refer to that in another part of the documentation though.